src/Symfony/Component/HttpKernel/HttpCache/Esi.php

```diff
@@ -89,16 +88,12 @@ public function process(Request $request, Response $response): Response
                 throw new \RuntimeException('Unable to process an ESI tag without a "src" attribute.');
             }
 
-            $chunks[$i] = sprintf('<?php echo $this->surrogate->handle($this, %s, %s, %s) ?>'."\n",
-                var_export($options['src'], true),
-                var_export($options['alt'] ?? '', true),
-                isset($options['onerror']) && 'continue' === $options['onerror'] ? 'true' : 'false'
-            );
-            ++$i;
-            $chunks[$i] = str_replace($this->phpEscapeMap[0], $this->phpEscapeMap[1], $chunks[$i]);
-            ++$i;
+            $chunks[$i] = $options['src']."\n".($options['alt'] ?? '')."\n".('continue' === ($options['onerror'] ?? ''));
+            $i += 2;
         }
-        $content = implode('', $chunks);
+        static $boundary;
```

Comment:
shouldn't this use an instance variable instead, so that separate instances can have separate boundaries ? Or maybe a normal local variable so that each processed response has a separate boundary.

And shouldn't we escape the boundary itself in case it appears in the content ?

Reply:
if you reuse it (especially in a long-running app server), you will lose the benefit of the randomness, potentially allowing to craft an attack by injecting the boundary in the output (as such setup would reuse the boundary long enough to try to guess it)

Reply:
alright, updated to rotate it